From 7929bf887c69ffdcbdfb525825bffba4c9e5d6e8 Mon Sep 17 00:00:00 2001
From: Romain Geissler <romain.geissler@amadeus.com>
Date: Fri, 10 May 2024 16:24:57 +0000
Subject: [PATCH] Allow option --no-tcp-fastopen to work on Linux kernels >=
 4.11.

* libwget/net.c (set_socket_options): Add check for tcp->tcp_fastopen.

Copyright-paperwork-exempt: Yes
---
 libwget/net.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/libwget/net.c b/libwget/net.c
index 8fc6d143..836649c0 100644
--- a/libwget/net.c
+++ b/libwget/net.c
@@ -640,9 +640,11 @@ static void set_socket_options(const wget_tcp *tcp, int fd)
 #endif
 
 #ifdef TCP_FASTOPEN_LINUX_411
-	on = 1;
-	if (setsockopt(fd, IPPROTO_TCP, TCP_FASTOPEN_CONNECT, (void *)&on, sizeof(on)) == -1)
-		debug_printf("Failed to set socket option TCP_FASTOPEN_CONNECT\n");
+	if (tcp->tcp_fastopen) {
+		on = 1;
+		if (setsockopt(fd, IPPROTO_TCP, TCP_FASTOPEN_CONNECT, (void *)&on, sizeof(on)) == -1)
+			debug_printf("Failed to set socket option TCP_FASTOPEN_CONNECT\n");
+	}
 #endif
 }
 
-- 
2.43.0

